### ISSUE TITLE
All students from all classes in the same csv sheet

### ISSUE THREAD

**Issue by TeacherAu** [al-caughey/Google-Meet-Attendance#93]
Something has changed since last week and this extension no longer gives a sheet with students present in ONE Meet, but, for example, if it's the 3rd Meet of the day, I get the students present from the first two, as well :( Any solutions, please?

**Comment by AlCaughey** [comment]
Are you using the same Meet for all classes?  I.e., the same 10 alpha id
and picking a new class from the drop down  list ?

Are you in the same Chrome sessions  or do you close Chrome between classes?

**Comment by TeacherAu** [comment]
I use Firefox and my classes are one after the other, so no, have never closed the browser after a class.
 I have a school account with 10 different groups/ classes. So I have 10 different permanent Meet links for each and use them according to our schedule...that’s why it seems weird to me... Plus, last week everything went perfectly...

**Comment by Eagle-Man** [comment]
Just to be clear, you do have a different class list in GMA for each class and you're switching between lists after each Meet?

**Comment by TeacherAu** [comment]
I have not uploaded any class lists in GMA (if that's what you meant). I
now have to reset the class list after every Meet, so as no to have the
names of the previous class mixed with the current one. As I explained, I
am a teacher and I have 10 different classes/ groups of students. I think
GMA could have its own list from the "People" list in Classroom, if need
be, or, simply remember all the names from EACH class's permanent Meet link
and tick those who are present, and add those who haven't attended before.
Also, I hope Meet will be improved soon, as it is not OK to give the power
to kick sb out to those who enter the Meet first, but only to those
registered as teachers. Thank you for your help! (Sorry if the stuff about
Meet doesn't go here!)

On 13 Nov 2020, at 22:38, Eagle-Man <notifications@github.com> wrote:

Just to be clear, you do have a different class list in GMA for each class
and you're switching between lists after each Meet?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/al-caughey/Google-Meet-Attendance/issues/93#issuecomment-727018668>,
or unsubscribe

**Comment by Eagle-Man** [comment]
@TeacherAu So, the problem is that you do not have class lists set up in the extension.  The GMA extension has almost exactly what you describe as its primary functionality.  If you set up class lists for each of your classes, then a report will be created for that class list when the Meet ends.  You just need to select the correct class list from the dropdown before you enter the Meet (or even when you're in it!).  Check out the "Getting Started..." help section in the extension by clicking the blue ? in the lower right corner.

This extension does not have access to your GCalendar, Classrooms, or other GSuite information for privacy and security reasons.  Also, no Meet links are permanent.  Even the ones attached to Calendar events will expire 90 days (?) after the last use.

Further, while GMA and Al have no say over how Meet works, Meet is a product for everyone, not just teachers, so they will likely not make it so that only "registered teachers" can be hosts of a Meet.  If you don't want students to be hosts, approach your organization about having the Google Admin disable creation of Meets for students.  This way, they cannot get into a Meet link until after it has been created by someone with authorization.  (not sure how this works with scheduled events in Calendar, though)

**Comment by TeacherAu** [comment]
Thank you so much for taking the time to explain! I guess I will set up class lists then for GMA.
Regarding permanent Meet links, I thought the ones under the class name in Google Classroom were permanent, but at least they are permanently there, which is very helpful :) We only use those according to our schedule, without scheduling each class in Calendar.
And, it seems right to me that anyone can be a host since it’s a product for everyone, I guess I just don’t like the fact that students are able to kick out teachers, as some of my colleagues have complained, so I will approach our IT teacher about it.
Again thank you for explaining everything! This was really helpful!

**Comment by Eagle-Man** [comment]
Ah, yes!  Classroom.  Sorry.  With Classroom, that one is more for education, so there may be something they can do, but I think it would have to be on the Meet end, which might make them rethink.

Regarding permanent Meet links...well, the actual room link/URL changes quite often with Classroom-sources Meets.  I believe it changes at least every time you disable and enable the Meet link, if not more often.  But the link that you see in Classroom is actually a “lookup” URL.  So they give the code/nickname to see if there’s already a Meet with that nickname in your organization; if there is, it returns the correct URL and forwards you there; if not, then if you have the ability to make a room, then it makes the room with that code/nickname.  This is why the first person in becomes the host - the room didn’t actually exist until someone went in.  You can even create your own rooms this way.  Just use a URL like meet.google.com/lookup/nickname .  Try it out.

If you would like to ensure you are the host all the time, begin by making Meet invisible to students.  Then, reset the Meet link (this will make sure no one connects to anything old).  Then, when you are ready to have a Meet, first you click the link in the Stream banner (teacher can still see this even when it’s invisible to students) to start the Meet, then go in to settings and make it visible to students.

**Comment by TeacherAu** [comment]
Just to let you know, I still can't save any of my classes. I only access
Meet from Classroom and, even though I tried to save my classes, the moment
I saved one and wanted to save a second one, I could not, no option to save
two classes.

**Comment by Eagle-Man** [comment]
There are no “save class list” options.  Class lists are automatically saved, just click outside of the extension window once you’ve finished editing the class.

How are you bc resting the class lists?  What happens when you try to follow the instructions for creating class lists?

**Comment by TeacherAu** [comment]
I go on Meet from Classroom, but there is no option " create"...or I'm just
not seeing it :(

**Comment by TeacherAu** [comment]
It's OK, I'll just keep resetting before each Meet and have the list with
everyone present from each class.

**Comment by Eagle-Man** [comment]
Check the instruction in the help section (? Icon) or in Al’s getting started YouTube video - you’ll also need to check out the 1.2.6 new features video as there’s one small change to setting up class lists from previous versions.

**Comment by TeacherAu** [comment]
Yeah, I looked everywhere last month. I don't have an 'Add' button in the
Class list dropdown. All I get is My Classes (and nothing there 'cause of
the reasons I've been mentioning) and Other Options - Manage, History,
Reset.

**Comment by Eagle-Man** [comment]
In previous versions, Add was where Manage is now.  But now classes are added through the Manage view.  Again, please see Al’s 1.2.6 update video on YouTube or in the Facebook group, where he walks through how to use this view.
